File: data_structures/programmers_unfinished_runner.cpp
/*
정확성  테스트
테스트 1 〉	통과 (0.01ms, 4.19MB)
테스트 2 〉	통과 (0.01ms, 4.18MB)
테스트 3 〉	통과 (0.20ms, 3.79MB)
테스트 4 〉	통과 (0.36ms, 4.05MB)
테스트 5 〉	통과 (0.37ms, 4.18MB)
효율성  테스트
테스트 1 〉	통과 (22.60ms, 17.9MB)
테스트 2 〉	통과 (35.62ms, 25.4MB)
테스트 3 〉	통과 (43.91ms, 30MB)
테스트 4 〉	통과 (48.75ms, 32.5MB)
테스트 5 〉	통과 (47.20ms, 32.5MB)
*/

#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    unordered_map<string, int> people;
    for (auto p : participant) people[p]++;
    
    for (auto c : completion) {
        people[c]--;
        if (people[c] == 0) people.erase(c);
    }
    
    answer = people.begin()->first;
    
    return answer;
}

File: data_structures/programmers_unfinished_runner.java
/*
정확성  테스트
테스트 1 〉	통과 (0.04ms, 73MB)
테스트 2 〉	통과 (0.07ms, 70.8MB)
테스트 3 〉	통과 (0.42ms, 80.3MB)
테스트 4 〉	통과 (0.65ms, 84.2MB)
테스트 5 〉	통과 (0.67ms, 77.2MB)
효율성  테스트
테스트 1 〉	통과 (72.56ms, 81.2MB)
테스트 2 〉	통과 (81.83ms, 89MB)
테스트 3 〉	통과 (83.28ms, 94.4MB)
테스트 4 〉	통과 (80.00ms, 96MB)
테스트 5 〉	통과 (81.53ms, 95.5MB)
*/

import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hm = new HashMap<>();
        for (String player : participant) hm.put(player, hm.getOrDefault(player, 0) + 1);
        for (String player : completion) hm.put(player, hm.get(player) - 1);

        for (String key : hm.keySet()) {
            if (hm.get(key) != 0) {
                answer = key;
                break;
            }
        }

        return answer;
    }
}
